compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1098,7 +1098,10 @@ object desugar {
           val firstDef =
             ValDef(tmpName, TypeTree(), matchExpr)
               .withSpan(pat.span.union(rhs.span)).withMods(patMods)
-          def selector(n: Int) = Select(Ident(tmpName), nme.selectorName(n))
+          val arity = vars.length
+          def selector(n: Int) =
```

Comment:
As `selector` is called in a large loop, I think it makes more sense for it to be a function val, and only check `arity < 22` once

Reply:
Done